src/pages/ProductDetail/ProductTab.js

```diff
@@ -1,62 +1,44 @@
-import React, { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import ProductInfo from './ProductInfo';
 import ProductReview from './ProductReview';
+
 import './ProductTab.scss';
 
-const ProductTab = () => {
-  const [detail, setDetail] = useState([]);
+const ProductTab = ({ detail }) => {
   const [count, setCount] = useState(1);
   const [price, setPrice] = useState(0);
   const [heart, setHeart] = useState('🖤');
   const [currentTab, setCurrentTab] = useState('info');
 
-  useEffect(() => {
-    fetch(`/data/productdetail/detaildata.json
-      `)
-      .then(res => res.json())
-      .then(res => setDetail(res));
-  }, []);
-
   const mappingObje = {
-    info: <ProductInfo />,
-    review: <ProductReview />,
-  };
-
-  const up = () => {
-    setCount(count + 1);
-  };
-
-  const down = () => {
-    count !== 1 && setCount(count - 1);
+    info: <ProductInfo info={detail[0]} />,
+    review: <ProductReview review={detail.slice(1, detail.length)} />,
   };
 
-  const buyClick = () => {
-    setPrice(detail[0].price * count);
-  };
-
-  const onHeart = () => {
-    heart === '🖤' ? setHeart('❤️') : setHeart('🖤');
-  };
+  const up = () => setCount(count + 1);
+  const down = () => count !== 1 && setCount(count - 1);
```

Comment:
여기는 함수형 업데이트를 사용하면 좋을 것 같네요.

Reply:
멘토님 이 부분은 잘 모르겠습니다...

Reply:
관훈님 이 부분은 해결되었을까요?
```suggestion
  const up = () => setCount(count => count + 1);
  const down = () => setCount(count => count !== 1 ? count - 1: 0);
```
으로 배치값을 이용하면 좋을 것 같다!라는 의견입니다.

Reply:
state 의 비동기 처리때문에 콜백함수로 최신화 해야 되는 상황이라고 생각하는데 맞을까요..? 근데 궁금한게 1씩 한번 증가 인데도 콜백함수로 처리 해야되는 건가요???